src/main/java/com/spotify/reaper/resources/view/NodesStatus.java

```diff
@@ -0,0 +1,208 @@
+      if(matcher.find()) {
+        severity = Optional.fromNullable(Double.parseDouble(matcher.group(3)));
+      }
+      
+      matcher = endpointLoadPattern.matcher(endpointString);
+      if(matcher.find()) {
+        load = Optional.fromNullable(new BigDecimal(matcher.group(3)).doubleValue());
+        totalLoad+=load.or(0.0);
+      }
+      
+      matcher = endpointReleasePattern.matcher(endpointString);
+      if(matcher.find()) {
+        releaseVersion = Optional.fromNullable(matcher.group(3));
+      }
+      
+      matcher = endpointHostIdPattern.matcher(endpointString);
+      if(matcher.find()) {
+        hostId = Optional.fromNullable(matcher.group(3));
+      }
+      
+      matcher = endpointTokensPattern.matcher(endpointString);
+      if(matcher.find()) {
+        tokens = Optional.fromNullable(matcher.group(2));
+      }
+      
+      
+      
+      EndpointState endpointState = new EndpointState(endpoint.or("Not available"), hostId.or("Not available"), dc.or("Not available"), rack.or("Not available"), status.or("Not available"), severity.or(0.0),
+          releaseVersion.or("Not available"), tokens.or("Not available"), load.or(0.0));
+      
+      endpoints.add(endpointState);
+    }
+    
+    Map<String, Map<String, List<EndpointState>>> endpointsByDcAndRack = Maps.newHashMap();
+    Map<String, List<EndpointState>> endpointsByDc = endpoints.stream().collect(Collectors.groupingBy(EndpointState::getDc, Collectors.toList()));
+    
+    for(String dc:endpointsByDc.keySet()) {
+      Map<String, List<EndpointState>> endpointsByRack = endpointsByDc.get(dc).stream().collect(Collectors.groupingBy(EndpointState::getRack, Collectors.toList()));
+      endpointsByDcAndRack.put(dc, endpointsByRack);
+    }
+    
+    GossipInfo gossipInfo = new GossipInfo(sourceNode ,endpointsByDcAndRack, totalLoad);
+    
+    return gossipInfo;
+  }
+
+  public class GossipInfo {
+    @JsonProperty
+    public final String sourceNode;
+    @JsonProperty
+    public final Map<String, Map<String, List<EndpointState>>> endpoints;
+    @JsonProperty
+    public final Double totalLoad;
+    
+    public GossipInfo(String sourceNode, Map<String, Map<String, List<EndpointState>>> endpoints, Double totalLoad) {
+      this.sourceNode = sourceNode;
+      this.endpoints = endpoints;
+      this.totalLoad = totalLoad;
+    } 
+  }
+  
+  public class EndpointState {
+    @JsonProperty
+    public final String endpoint;
+    @JsonProperty
+    public final String hostId;
+    @JsonProperty
+    public final String dc;
+    @JsonProperty
+    public final String rack;
+    @JsonProperty
+    public final String status;
+    @JsonProperty
+    public final Double severity;
+    @JsonProperty
+    public final String releaseVersion;
+    @JsonProperty
+    public final String tokens;
+    @JsonProperty
+    public final Double load;
+    
+    public EndpointState(String endpoint, String hostId, String dc, String rack, String status, Double severity,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)